diff --git a/src/process.c b/src/process.c
index 933db7a..a35a41a 100644
--- a/src/process.c
+++ b/src/process.c
@@ -322,8 +322,15 @@ struct apk_istream *apk_process_istream(char * const* argv, struct apk_out *out,
 	r = apk_process_init(&pis->proc, apk_last_path_segment(argv[0]), logpfx, out, NULL);
 	if (r != 0) goto err;
 
-	r = apk_process_spawn(&pis->proc, argv[0], argv, NULL);
-	if (r != 0) goto err;
+	pid_t pid = apk_process_fork(&pis->proc);
+	if (pid < 0) {
+		r = errno;
+		goto err;
+	}
+	if (pid == 0) {
+		execvp(argv[0], argv);
+		_exit(127);
+	}
 
 	return &pis->is;
 err:
