diff --git a/src/io_url_wget.c b/src/io_url_wget.c
index d2098f1..ffbb0d7 100644
--- a/src/io_url_wget.c
+++ b/src/io_url_wget.c
@@ -20,8 +20,10 @@ struct apk_istream *apk_io_url_istream(const char *url, time_t since)
 
 	argv[i++] = "wget";
 	argv[i++] = "-q";
-	argv[i++] = "-T";
-	argv[i++] = wget_timeout;
+	if (wget_timeout[0] != '\0') {
+		argv[i++] = "-T";
+		argv[i++] = wget_timeout;
+	}
 	if (wget_no_check_certificate) argv[i++] = "--no-check-certificate";
 	argv[i++] = (char *) url;
 	argv[i++] = "-O";
