diff --git a/libfetch/meson.build b/libfetch/meson.build
index eac5b31..4052d38 100644
--- a/libfetch/meson.build
+++ b/libfetch/meson.build
@@ -18,7 +18,7 @@ libfetch = static_library(
 	c_args: libfetch_cargs,
 	dependencies: [
 		libportability_dep.partial_dependency(compile_args: true, includes: true),
-		crypto_dep.partial_dependency(compile_args: true, includes: true)
+		crypto_dep
 	],
 )
 
